### DISCUSSION TITLE
The future of crouton: an eerie silence

### DISCUSSION THREAD

**Discussion by CroutonIsFun** [dnschneid/crouton#4967] [General]
In March of 2023, a component of the Chromium open source project was removed. For the technically inclined, here is the relevant link:

[autotools removed](https://chromium.googlesource.com/chromiumos/third_party/adhd/+/879fe7a51784eb07d1b85473d8927050fb000f87)

The component was listed in the 3rd party area of the website. I don't know why the component, known as "autotools",  was removed. The component enables the build, make, compile (I don't know the difference) of _cras_, the ChRome Audio Server. Once this tool was removed, it became impossible to install crouton without errors.

Removing autotools has not affected the development of ChromeOS, but it has presented a nearly unsolvable issue for the crouton devs. To get around this issue, @dnschneid pushed a branch will removes the audio dependency. You can now install crouton like this:

`sudo CROUTON_BRANCH=silence crouton -r xenial -t xfce`

Your xfce desktop will be installed! However, there will be **no sound**! On the plus side, there is no downloading of i386 packages, which were needed so that autotools could compile cras. So your install will go much more quickly.

Crouton has many use cases besides audio, so hopefully, this will help numbers of people who wish to use crouton alongside ChromeOS.

Is there anyway to get sound after the install? The answer is a definite maybe. I was never able to get the sound working using the built-in sound hardware. But I managed to get sound working by using a simple USB sound device. The device has 3.5 mm audio ports. I run an audio cable from the green port to my boombox AUX port. There is also a mic port, which I don't use, because I am already using a "Blue Snowball" high quality mic, which works fine without additional issues.

To get sound after the above install, you must follow these steps:

Add your username to the hwaudio group, e.g., `sudo nano /etc/group`

This can be done from the command line, but to me it's easier to just use an editor. Plus, I need to add my username to the cd-rom group. Exit the chroot and start again.

Install alsa-utils: `sudo apt install alsa-utils`

Create a file called `.asoundrc` which specifies the default sound card. Here is what mine looks like:

pcm.!default {
type hw
card Device }

ctl.!default {
type hw
card Device}

You can get a list of sound cards with `cat /proc/asound/cards`

The external device I own is made by a company called "Pluggable". I couldn't find it on Amazon, but here is a close match:

[SABRENT USB External Stereo Sound Adapter](https://www.amazon.com/Sabrent-External-Adapter-Windows-AU-MMSA/dp/B00IRVQ0F8/)

Disclaimer: I don't have an Amazon seller account, and make nothing off this link. I am not associated in any manner with the seller.

Edit: the USB device only costs about $7 or $8, depending on which model you choose. The above product is very popular on Amazon, because it is Plug-and-Play: specifically, for Windows and Mac, but it will work for Linux also.

**Comment by Taxrosdev** [comment]
My question is: Is it possible to also use pipewire or pulse?

**Reply by CroutonIsFun** [reply]
@Eaterminer

If you can get sound at all, you should be able to install and use pulseaudio.

Pipewire, if I remember, was needed several years ago by Linux installs so you could watch Netflix. This is no longer an issue. Chromebook is a great client for watching Netflix and other on-demand streaming services.

**Reply by Taxrosdev** [reply]
Expanding on pipewire.
Pipewire is an alternative for pulse and alsa, and IMO it's better. It's just an opinion tho

**Reply by CroutonIsFun** [reply]
Cool

**Comment by Qwer-TeX** [comment]
Thank You
